tests/testthat/test-lengths_linter.R

```diff
@@ -0,0 +1,35 @@
+test_that("lengths_linter skips allowed usages", {
+  # TODO(michaelchirico): this probably should be lengths(), but let it pass for now.
+  #   Seen twice as a usage like lapply(x, length) > 0 which does as.integer()
+  #   conversion and thus is equivalent to sapply(x, length). Check whether
+  #   there are any real use cases for lapply(x, length)...
+  expect_lint("lapply(x, length)", NULL, lengths_linter())
```

Comment:
Could safely lint this (and map) suggesting `as.list(lengths(...))` instead. Can be a follow-up though.

Reply:
True, and I'm impressed just how much faster it is than even `vapply(x, length)`. That evaluation overhead is a killer.

```
library(microbenchmark)
l = replicate(1e5, sample(10, sample(10, 1)), simplify=FALSE)
microbenchmark(as.list(lengths(l)), sapply(l, length), vapply(l, length, 1L))
# Unit: milliseconds
#                   expr       min        lq      mean    median        uq       max neval
#    as.list(lengths(l))  4.846123  5.423464  7.144133  5.637852  5.988355  29.90146   100
#      sapply(l, length) 37.334026 42.108947 50.622672 49.059130 50.951532 113.20739   100
#  vapply(l, length, 1L) 33.147643 36.679405 44.902168 43.278030 45.429923 110.50852   100

l = replicate(1e6, sample(10, sample(10, 1)), simplify=FALSE)
microbenchmark(as.list(lengths(l)), sapply(l, length), vapply(l, length, 1L))
# Unit: milliseconds
#                   expr       min       lq     mean    median       uq      max neval
#    as.list(lengths(l))  53.26755  58.4670 127.9654  90.16496 127.2416 436.8011   100
#      sapply(l, length) 431.83550 474.3372 573.0207 523.75971 628.0669 854.3293   100
#  vapply(l, length, 1L) 367.80554 403.4373 454.7605 421.15209 455.9736 748.4410   100
```

Reply:
#1570